build-script.py

```diff
@@ -364,6 +364,8 @@ def main():
         sys.exit(1)
 
     try:
+        _environ = dict(os.environ)
+        os.environ['SWIFT_EXEC'] = args.swiftc_exec
```

Comment:
Instead of mutating `os.environ`, you should use `env=` argument for `subprocess.check_call`.

Reply:
Good point! I'll make another PR.